tests/components/xiaomi_miio/test_button.py

```diff
@@ -0,0 +1,104 @@
+"""The tests for the xiaomi_miio button component."""
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from homeassistant.components.button import (
+    DOMAIN,
+    SERVICE_PRESS,
+)
+from homeassistant.components.xiaomi_miio.const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    CONF_MAC,
+    DOMAIN as XIAOMI_DOMAIN,
+    MODELS_VACUUM,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_HOST,
+    CONF_MODEL,
+    CONF_TOKEN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt as dt_util
+
+from . import TEST_MAC
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=True)
+async def setup_test(hass: HomeAssistant):
+    """Initialize test xiaomi_miio for button entity."""
+
+    mock_vacuum = MagicMock()
+
+    with patch(
+        "homeassistant.components.xiaomi_miio.get_platforms",
+        return_value=[
+            Platform.BUTTON,
+        ],
+    ), patch("homeassistant.components.xiaomi_miio.RoborockVacuum") as mock_vacuum_cls:
+        mock_vacuum_cls.return_value = mock_vacuum
+        yield mock_vacuum
+
+
+async def test_vacuum_button_params(hass: HomeAssistant) -> None:
+    """Test the initial parameters of a vacuum button."""
+
+    entity_name = "test_vacuum"
+    entity_id = await setup_component(hass, entity_name)
+
+    state = hass.states.get(entity_id + "_reset_main_brush")
+    assert state
+    assert state.state == "unknown"
+
+
+async def test_vacuum_button_press(hass: HomeAssistant) -> None:
+    """Test pressing a vacuum button."""
+
+    entity_name = "test_vacuum"
+    entity_id = await setup_component(hass, entity_name)
```

Comment:
The entity name isn't used multiple times... so assigning it seems unneeded.

```suggestion
    entity_id = await setup_component(hass, "test_vacuum")
```

Reply:
Same as https://github.com/home-assistant/core/pull/91483#discussion_r1174441079